bench/index.js

```diff
@@ -0,0 +1,139 @@
+        simple_select_large: require('./tests/simple_select_large'),
+        simple_select_large_zoomed: require('./tests/simple_select_large_zoomed'),
+
+        direct_select_small: require('./tests/direct_select_small'),
+        direct_select_small_zoomed: require('./tests/direct_select_small_zoomed'),
+        direct_select_large: require('./tests/direct_select_large'),
+        direct_select_large_zoomed: require('./tests/direct_select_large_zoomed'),
+
+        draw_line_string_small: require('./tests/draw_line_string_small'),
+        draw_line_string_large: require('./tests/draw_line_string_large'),
+        draw_line_string_large_zoomed: require('./tests/draw_line_string_large_zoomed'),
+
+        draw_polygon_small: require('./tests/draw_polygon_small'),
+        draw_polygon_large: require('./tests/draw_polygon_large'),
+        draw_polygon_large_zoomed: require('./tests/draw_polygon_large_zoomed'),
+
+        draw_land_polygon_small: require('./tests/draw_land_polygon_small'),
+        draw_land_polygon_large: require('./tests/draw_land_polygon_large'),
+
+        draw_urban_areas_polygon_small: require('./tests/draw_urban_areas_polygon_small'),
+        draw_urban_areas_polygon_large: require('./tests/draw_urban_areas_polygon_large'),
+
+        draw_point_small: require('./tests/draw_point_small'),
+        draw_point_large: require('./tests/draw_point_large'),
+        draw_point_large_zoomed: require('./tests/draw_point_large_zoomed'),
+    };
+
+    var pathnameArray = location.pathname.split('/');
+    var benchmarkName = pathnameArray[pathnameArray.length - 1] || pathnameArray[pathnameArray.length - 2];
+
+    var testDiv = document.getElementById('tests');
+    var tests = Object.keys(benchmarks);
+
+    var innerHTML = '';
+
+    tests.forEach(function(test) {
+        innerHTML += '<div class="test">';
+            innerHTML += '<a href="/bench/'+test+'">'+test+'</a>';
+        innerHTML += '</div>';
+        if (test === benchmarkName) {
+            console.log('match', test);
+            innerHTML += '<div id="logs"></div>';
+        }
+    });
+
+    testDiv.innerHTML = innerHTML;
+
+    log('dark', 'please keep this window in the foreground and close the debugger');
+
+    var createBenchmark = benchmarks[benchmarkName];
+    if (!createBenchmark) {
+        log('dark', benchmarkName+' is not a valid test name');
+        return;
+    }
+
+    var benchmark = createBenchmark({
+        accessToken: getAccessToken(),
+        createMap: createMap
+    });
+
+    benchmark.on('log', function(event) {
+        log(event.color || 'blue', event.message);
+        scrollToBottom();
+    });
+
+    benchmark.on('pass', function(event) {
+        log('green', '<strong class="prose-big">' + event.message + '</strong>');
+        scrollToBottom();
+    });
+
+    benchmark.on('fail', function(event) {
+        log('red', '<strong class="prose-big">' + event.message + '</strong>');
+        scrollToBottom();
+    });
```

Comment:
What does it mean for a benchmark to "pass" or "fail"? Don't you want a metric that you can work to improve over time?